### ISSUE TITLE
Suggest: Option to turn autocomplete On / Off

### ISSUE THREAD

**Issue by SkunkerAG** [ixti/redmine_tags#157]
Hi,

your plugin is great, but there is a problem when using  a lot of tags in one Project(20+).
The autocorrection dropdown list does sometimes not go away on page close or opens to early when starting editing the ticket. 

It would be great if a turn on/off option for autocorrection is added. 

Thank you

**Comment by SkunkerAG** [comment]
THe problem occurs in  Internet Explorer 11 with:

Environment:
  Redmine version                3.2.1.stable
  Ruby version                   2.3.1-p112 (2016-04-26) [x86_64-linux-gnu]
  Rails version                  4.2.7.1
  Environment                    production
  Database adapter               Mysql2

SCM:
  Git                            2.7.4
  Filesystem                     

Redmine plugins:
  issue_charts                   1.3.0
  quick_view                     0.3.0
  redmine_agile                  1.4.1
  redmine_didyoumean             1.2.0
  redmine_editauthor             0.9.0
  redmine_issue_badge            0.0.3
  redmine_ldap_sync              2.0.8.devel
  redmine_lightbox2              0.2.7
  redmine_people                 1.2.0
  redmine_tags                   3.1.1
  redmine_theme_changer          0.2.0

**Comment by marius-balteanu** [comment]
Thanks for reporting the issue.  We'll try to reproduce and fix the issue instead of adding an option to disable/enable the autocomplete feature.

**Comment by AstraSerg** [comment]
I agree with marius. Disable/enable autocomplete feature is not needed. It will be hard to use this plugin without an autocomplition.

**Comment by sdwolf** [comment]
I will look into this over the weekend

**Comment by marius-balteanu** [comment]
The autocomplete dropdown can be very long on a project with many tags (see below screenshot). I think we can limit the results to 20 (for example).

 @PowerKiKi, @sdwolf, let me know what do you think about this change.


<img width="1274" alt="screen shot 2016-12-18 at 22 20 04" src="https://cloud.githubusercontent.com/assets/5037739/21296255/40bf5f0a-c570-11e6-9e47-b4522d0b2ff6.png">

**Comment by PowerKiKi** [comment]
I am not against limiting the result list, but it does not really seem to address the original issue.

Especially I am having a hard time imagining how a page element may "not go away on page close". If the page is actually closed, I really don't see how anything could linger around.

Also I don't think "opens too early" can be a problem. On the contrary I see reactivity and fast feedback as a (great) feature.

IMHO I wouldn't do much until we understand what is the actual root issue. But again, I don't mind all that much to limit the list length.

**Comment by ixti** [comment]
I think limiting to 20 tags is fair enough.

**Comment by SkunkerAG** [comment]
Hi there,

Thanks for your help. Limiting is for me a good idea, but not solving that problem. 

>Also I don't think  "opens too early" can be a problem. 
Sadly it can :( It pops up when you're not even editing the ticket, while you're just watching it. Not even hitting F5 (reload) changes this behavior.

> "not go away on page close":
 I wanted to say with that, if you close editing mode, it does not go away. The list, with 20+ results stays open. Beside that , if you have many results, and the list opens it expands down, and out of the screen. This is why i think limiting is a good idea, too.

But as I said in this post, it's not solving the "showing up in unexpected situations" problem.

plus: we found out, that it's all good in Firefox... 

hope to be helpfull :)

**Comment by PowerKiKi** [comment]
It seems that there are several different issues, that should probably have been reported separately and with more details. Especially if the "opens too early" is experienced only on a specific browser you should let us know what browsers you tried, and which one did/didn't worked.

**Comment by SkunkerAG** [comment]
Sorry for reporting in one error.  That's because I first suggested a 'turn-it-off'-feature.

The browser that has problems with the popup is the one i wrote above: IE 11. Here are more Informations: 

Internet Explorer 11
  Version 11.0.9600.18524
  Updateversion 11.0.37 (KB3197655)

Firefox 50.0.1 Okay
Chrome: Okay
that's all versions I can test here.

this happened e.g. , when I switched the tracker:

![image](https://cloud.githubusercontent.com/assets/24564989/21319739/b86ba146-c60e-11e6-876e-d0f7b6187c6d.png)

**Comment by sdwolf** [comment]
@SkunkerAG I am unable to reproduce this on IE11, but I have a newer version: `11.0.14393.0` and also running on Windows 10. Not sure if this this bug is related to the browser or some other JavaScript library from another plugin. Can you please try to reproduce this bug on your system with a fresh Redmine instance and only the `redmine_tags` plugin?.

Also I am planning to switch the `tag-it` library with a newer one, hopefully this may resolve more bugs (even the one related to Cyrillic characters)
